test/parallel/test-timers-immediate.js

```diff
@@ -14,17 +11,19 @@ setImmediate(common.mustCall(function() {
 
 const immediateB = setImmediate(common.mustNotCall());
 
-setImmediate(function(x, y, z) {
-  immediateC = [x, y, z];
-}, 1, 2, 3);
+for (let n = 1; n <= 5; n++) {
+  const argsExpected = new Array(n).fill(0).map((_, i) => i);
+  let argsReceived;
+
+  const callback = common.mustCall(function() {
+    argsReceived = Array.from(arguments);
+  });
 
-setImmediate(function(x, y, z, a, b) {
-  immediateD = [x, y, z, a, b];
-}, 1, 2, 3, 4, 5);
+  setImmediate.apply(null, [callback].concat(argsExpected));
 
-process.on('exit', function() {
-  assert.deepStrictEqual(immediateC, [1, 2, 3], 'immediateC args should match');
-  assert.deepStrictEqual(immediateD, [1, 2, 3, 4, 5], '5 args should match');
-});
+  process.on('exit', function() {
```

Comment:
Perhaps we can remove this event handler and just move the assertion to inside `callback`?